modules/javafx.graphics/src/main/java/javafx/css/TransitionEvent.java

```diff
@@ -0,0 +1,117 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javafx.css;
+
+import javafx.animation.Transition;
+import javafx.event.Event;
+import javafx.event.EventType;
+import javafx.scene.Node;
+import javafx.util.Duration;
+
+/**
+ * An event that signals the creation, beginning, completion and cancellation of implicit CSS transitions.
+ * <p>
+ * Note that this event is not raised for explicit transitions that are created using the {@link Transition} class.
+ *
+ * @since 22
+ */
+public final class TransitionEvent extends Event {
+
+    private static final long serialVersionUID = 20220820L;
+
+    /**
+     * Common supertype for all {@code TransitionEvent} types.
+     */
+    public static final EventType<TransitionEvent> ANY =
+            new EventType<>(Event.ANY, "TRANSITION");
+
+    /**
+     * This event occurs when a transition has been created and added to the list of running
+     * transitions of a {@link Node}.
+     */
+    public static final EventType<TransitionEvent> RUN =
+            new EventType<>(TransitionEvent.ANY, "TRANSITION_RUN");
+
+    /**
+     * This event occurs when a running transition enters its active period, which happens
+     * at the end of the delay phase.
+     */
+    public static final EventType<TransitionEvent> START =
+            new EventType<>(TransitionEvent.ANY, "TRANSITION_START");
+
+    /**
+     * This event occurs when a running transition has reached the end of its active period.
+     * If the transition is cancelled prior to reaching the end of its active period, this
+     * event does not occur.
+     */
+    public static final EventType<TransitionEvent> END =
+            new EventType<>(TransitionEvent.ANY, "TRANSITION_END");
+
+    /**
+     * This event occurs when a running transition was cancelled before it has reached the
+     * end of its active period.
+     */
+    public static final EventType<TransitionEvent> CANCEL =
+            new EventType<>(TransitionEvent.ANY, "TRANSITION_CANCEL");
+
+    private final StyleableProperty<?> property;
+    private final Duration elapsedTime;
+
+    /**
+     * Creates a new instance of the {@code TransitionEvent} class.
+     *
+     * @param eventType the event type
+     * @param property the {@code StyleableProperty} that is targeted by the transition
+     * @param elapsedTime the time that has elapsed since the transition has entered its active period
```

Comment:
Can property be `null`?  Any restrictions on `elapsedTime`?

Reply:
Added a `@throws` tag.
Interestingly, `Event` doesn't specify or assert that its `eventType` parameter is non-null. Maybe we should investigate this further.

Reply:
`eventType` is not used for a lot of things, aside from deciding which list of handlers to pick.  In theory, you could have a `null` `EventType`, and register an event handler on the `null` type; except that most `addEventHandler` methods don't allow this -- they check that the event type is not `null`.

It probably would be good to reject `null` when creating an `Event`, but that's not for this PR.

Reply:
I added a null check for `eventType` on the `TransitionEvent` class.